src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3039 @@
+                EngineBleedAirSystem::new(
+                    context,
+                    1,
+                    ElectricalBusType::DirectCurrentEssentialShed,
+                ),
+                EngineBleedAirSystem::new(context, 2, ElectricalBusType::DirectCurrent(2)),
+            ],
+            cross_bleed_valve: CrossBleedValve::new(1.),
+            fadec: FullAuthorityDigitalEngineControl::new(context),
+            engine_starter_valve_controllers: [
+                EngineStarterValveController::new(1),
+                EngineStarterValveController::new(2),
+            ],
+            apu: CompressionChamber::new(Volume::new::<cubic_meter>(5.)),
+            apu_bleed_air_valve: DefaultValve::new_closed(),
+            apu_bleed_air_controller: ApuCompressionChamberController::new(context),
+            green_hydraulic_reservoir_with_valve: PneumaticContainerWithConnector::new(
+                VariableVolumeContainer::new(
+                    Volume::new::<gallon>(8.),
+                    Pressure::new::<psi>(14.7),
+                    ThermodynamicTemperature::new::<degree_celsius>(15.),
+                ),
+            ),
+            blue_hydraulic_reservoir_with_valve: PneumaticContainerWithConnector::new(
+                VariableVolumeContainer::new(
+                    Volume::new::<gallon>(15.),
+                    Pressure::new::<psi>(14.7),
+                    ThermodynamicTemperature::new::<degree_celsius>(15.),
+                ),
+            ),
+            yellow_hydraulic_reservoir_with_valve: PneumaticContainerWithConnector::new(
+                VariableVolumeContainer::new(
+                    Volume::new::<gallon>(10.),
+                    Pressure::new::<psi>(14.7),
+                    ThermodynamicTemperature::new::<degree_celsius>(15.),
+                ),
+            ),
+            packs: [PackComplex::new(context, 1), PackComplex::new(context, 2)],
+        }
+    }
+
+    pub(crate) fn update(
+        &mut self,
+        context: &UpdateContext,
+        engines: [&(impl EngineCorrectedN1 + EngineCorrectedN2); 2],
+        overhead_panel: &mut A320PneumaticOverheadPanel,
+        engine_fire_push_buttons: &impl EngineFirePushButtons,
+        hydraulics: &A320Hydraulic,
+    ) {
+        self.apu.update(&self.apu_bleed_air_controller);
+
+        for bleed_monitoring_computer in self.bleed_monitoring_computers.iter_mut() {
+            bleed_monitoring_computer.update(
+                context,
+                &self.engine_systems,
+                self.apu_bleed_air_valve.is_open(),
+                overhead_panel,
+                engine_fire_push_buttons,
+                self.cross_bleed_valve.is_open(),
+                &self.fadec,
+            );
+
+            // I am not exactly sure if both BMCs should actually control this valve all the time.
+            self.cross_bleed_valve
+                .update_open_amount(&bleed_monitoring_computer.main_channel);
```

Comment:
Oh, that's actually a good suggestion for the other signals as well. I'll have to look into it again, but I think this will allow me to get rid of the whole `channel_for_engine` stuff.